spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/Topology.scala

```diff
@@ -18,30 +18,24 @@ package com.intel.analytics.bigdl.nn.keras
 
 import com.intel.analytics.bigdl.nn.Graph._
 import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
-import com.intel.analytics.bigdl.nn.{Graph, GraphSerializable, StaticGraph, Sequential => TSequential}
+import com.intel.analytics.bigdl.nn.{Container, Graph, GraphSerializable, StaticGraph, Sequential => TSequential}
 import com.intel.analytics.bigdl.serialization.Bigdl.BigDLModule
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
 import com.intel.analytics.bigdl.utils.serializer._
-import com.intel.analytics.bigdl.utils.{Shape, Util}
 
+import scala.collection.JavaConverters._
 import scala.reflect.ClassTag
 
 class Model[T: ClassTag](private val _inputs : Seq[ModuleNode[T]],
       private val _outputs : Seq[ModuleNode[T]])(implicit ev: TensorNumeric[T])
-  extends StaticGraph[T](_inputs, _outputs, None, false) {
-
-  Util.excludeNotKeras(inputs.map(_.element))
-  Util.excludeNotKeras(outputs.map(_.element))
+  extends StaticGraph[T](_inputs, _outputs, None) {
 
   this.inputShapeValue = Shape(inputs.map{n => n.element.getInputShape()}.toList)
 
-  this.outputShapeValue = Array(outputs.map{_.element.getOutputShape()}: _*)
-
-  isBuilt = true
-
-  override private[bigdl] def isCompatibleWithKeras(): Boolean = true
+  this.outputShapeValue = Shape(outputs.map{_.element.getOutputShape()}.toList)
 
-  override private[bigdl] def isCompatibleWithTorch(): Boolean = false
+  override private[bigdl] def isKerasStyle(): Boolean = true
 
   override def computeOutputShape(inputShape: Shape): Shape = {
```

Comment:
What if inputShape is not compatible with this.inputShapeValue?

Reply:
There's a hooker: `checkWithCurrentInputShape` for this
``` scala
  private def checkWithCurrentInputShape(calcInputShape: Shape): Unit = {
    if (getInputShape() != null) {
      val withoutBatchInputShape = KerasLayer.removeBatch(getInputShape())
      val withoutBatchCalcInputShape = KerasLayer.removeBatch(calcInputShape)
      require(withoutBatchInputShape == withoutBatchCalcInputShape,
        s"InputShape from constructor ${withoutBatchInputShape}" +
          s"should be the same with the calculated inputShape: ${withoutBatchCalcInputShape}")
    }
  }

  override def inferShape(calcInputShape: Shape): Shape = {
    checkWithCurrentInputShape(calcInputShape)
    super.inferShape(calcInputShape)
  }
```

Reply:
These are defined in KerasLayer, but Model is not a KerasLayer